tools/CampaignConfigurationCheck/CampaignName.mjs

```diff
@@ -0,0 +1,15 @@
+import chalk from 'chalk';
+
+export default class CampaignName {
+	/**
+	 * @param {string} campaignName
+	 */
+	constructor( campaignName ) {
+		const testNumberRegexResult = campaignName.match( /_(\d{2})/ );
+		if ( testNumberRegexResult === null ) {
+			console.warn( `${ chalk.red( 'CAMPAIGN CONFIGURATION ERROR:' ) } Cannot parse test number ( "_dd" format) for ${ campaignName } ` );
```

Comment:
When I described these objects I was thinking of "simple value objects", without logging side effects and dependencies on console libraries. Could you throw exceptions instead and catch them when constructing? It'll be even less code since you have the `process.exit` in one place.